bench/app.py

```diff
@@ -171,42 +170,80 @@ def get(self):
 	def remove(self):
 		active_app_path = os.path.join("apps", self.repo)
 		archived_path = os.path.join("archived", "apps")
-		archived_name = get_available_folder_name(
-			f"{self.repo}-{date.today()}", archived_path
-		)
+		archived_name = get_available_folder_name(f"{self.repo}-{date.today()}", archived_path)
 		archived_app_path = os.path.join(archived_path, archived_name)
 		log(f"App moved from {active_app_path} to {archived_app_path}")
 		shutil.move(active_app_path, archived_app_path)
 
 	@step(title="Installing App {repo}", success="App {repo} Installed")
-	def install(self, skip_assets=False, verbose=False, restart_bench=True):
+	def install(self, skip_assets=False, verbose=False, resolved=False, restart_bench=True):
 		import bench.cli
 		from bench.utils.app import get_app_name
 
 		verbose = bench.cli.verbose or verbose
 		app_name = get_app_name(self.bench.name, self.repo)
-
-		# TODO: this should go inside install_app only tho - issue: default/resolved branch
-		setup_app_dependencies(
-			repo_name=self.repo,
-			bench_path=self.bench.name,
-			branch=self.tag,
-			verbose=verbose,
-			skip_assets=skip_assets,
-		)
+		if not resolved and self.repo != "frappe":
+			click.secho(
+				f"Ignoring dependencies of {self.name} to install dependencies use --resolve-deps",
+				fg="yellow",
+			)
 
 		install_app(
 			app=app_name,
+			tag=self.tag,
 			bench_path=self.bench.name,
 			verbose=verbose,
 			skip_assets=skip_assets,
-			restart_bench=restart_bench
+			restart_bench=restart_bench,
+			resolution = self.local_resolution
 		)
 
+	@step(title="Cloning and installing {repo}", success="App {repo} Installed")
+	def install_resolved_apps(self, *args, **kwargs):
+		self.get()
+		self.install(*args, **kwargs, resolved=True)
+
 	@step(title="Uninstalling App {repo}", success="App {repo} Uninstalled")
 	def uninstall(self):
 		self.bench.run(f"{self.bench.python} -m pip uninstall -y {self.repo}")
 
+	def _get_dependencies(self):
```

Comment:
1. Handle `on_disk`
2. Fix this

```python
>>> a = App("./apps/healthcare")
>>> a._get_dependencies()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/gavin/bench/bench/app.py", line 216, in _get_dependencies
    required_apps = eval(lines[0].strip("required_apps").strip().lstrip("=").strip())
IndexError: list index out of range
>>>
```

Reply:
fixed by [0b862c0](https://github.com/frappe/bench/pull/1257/commits/0b862c07457c9914840bd0ce87a90306fec37456)